PGM/src/main/java/tc/oc/pgm/shop/ShopInterface.java

```diff
@@ -0,0 +1,162 @@
+package tc.oc.pgm.shop;
+
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.chat.TranslatableComponent;
+import org.bukkit.entity.Player;
+import tc.oc.commons.bukkit.gui.Interface;
+import tc.oc.commons.bukkit.gui.buttons.Button;
+import tc.oc.commons.bukkit.gui.interfaces.ChestInterface;
+import tc.oc.commons.bukkit.util.ItemCreator;
+import tc.oc.pgm.PGMTranslations;
+import tc.oc.pgm.match.Match;
+import tc.oc.pgm.match.MatchPlayer;
+import tc.oc.pgm.shop.purchasable.Purchasable;
+import tc.oc.pgm.shop.purchasable.PurchasableSet;
+import tc.oc.pgm.shop.strategy.PaymentStrategy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * GUI that is created from the items in a {@link Shop} that can be interacted with.
+ */
+public class ShopInterface extends ChestInterface {
+    private static ShopInterface instance;
+
+    final MatchPlayer player;
+    final Shop shop;
+    final PurchaseTracker tracker;
+    final Match match;
+
+    public ShopInterface(MatchPlayer player, Shop shop) {
+        super(player.getBukkit(), new ArrayList<>(), shop.rows * 9, shop.getTitle(), getInstance());
+        this.player = player;
+        this.shop = shop;
+        this.tracker = shop.tracker;
+        this.match = player.getMatch();
+        updateButtons();
+    }
+
+    public static ShopInterface getInstance() {
```

Comment:
Literally just copied this from another interface class.
See `TokenPurchaseInterface`